File: looking/julia/padovan.jl
# See https://en.wikipedia.org/wiki/Padovan_sequence
# Inspired by https://math.stackexchange.com/questions/3798107/is-there-a-better-way-to-calculate-padovan-sequence-than-on

padovan(n) = ([0 1 0; 0 0 1; 1 1 0]^n)[3, 2]

expressions = [
    :(padovan(1))
    :(padovan(2))
    :(padovan(3))
    :(padovan(4))
    :(padovan(5))
    :(padovan(6))
    :(padovan(7))
    :(padovan(8))
    :(padovan(9))
    :(padovan(10))
    :(padovan(91))
    :(padovan(big(283)))
    :(padovan(big(3674)))
]

for expr in expressions
    println("[] :julia-result \"", escape_string(string(expr)), " = ", eval(expr), "\".")
end

File: let-prolog/input/padovan.pl
% See https://en.wikipedia.org/wiki/Padovan_sequence

:- op(1200, xfx, :+).

padovan(A, B) :-
    padovan(A, 0, 1, 1, B).

padovan(0, A, _, _, A).
padovan(1, _, A, _, A).
padovan(2, _, _, A, A).
padovan(A, B, C, D, E) :-
    A > 2,
    F is A-1,
    G is B+C,
    padovan(F, C, D, G, E).

plastic_ratio(A, B) :-
    padovan(A, C),
    D is A+1,
    padovan(D, E),
    B is E/C.

% query
true :+ padovan(1, _).
true :+ padovan(2, _).
true :+ padovan(3, _).
true :+ padovan(4, _).
true :+ padovan(5, _).
true :+ padovan(91, _).
true :+ padovan(283, _).
true :+ padovan(3674, _).
true :+ plastic_ratio(1, _).
true :+ plastic_ratio(10, _).
true :+ plastic_ratio(100, _).
true :+ plastic_ratio(1000, _).
